arbnode/node.go

```diff
@@ -570,120 +573,147 @@ func getDAS(
+		}
+
+		// Register external DA client as both reader and validator
+		result, err := externalDAClient.GetSupportedHeaderBytes().Await(ctx)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to get supported header bytes from external DA client[%d]: %w", i, err)
 		}
-		// Only allow dawriter if batchposter is enabled
-		withDAWriter = config.DAProvider.WithWriter && config.BatchPoster.Enable
-	} else if config.DataAvailability.Enable {
+		for _, hb := range result.HeaderBytes {
+			if err := dapRegistry.Register(hb, externalDAClient, externalDAClient); err != nil {
+				return nil, nil, nil, fmt.Errorf("failed to register DA provider[%d]: %w", i, err)
+			}
+		}
+	}
+
+	// Create AnyTrust DA provider if enabled (can coexist with external DA)
+	if config.DataAvailability.Enable {
+		// Map deprecated BatchPoster.MaxSize to DataAvailability.MaxBatchSize for backward compatibility
+		if config.BatchPoster.MaxSize != 0 && config.DataAvailability.MaxBatchSize == das.DefaultDataAvailabilityConfig.MaxBatchSize {
+			log.Warn("Using deprecated batch-poster.max-size for AnyTrust max batch size; please migrate to data-availability.max-batch-size")
+			config.DataAvailability.MaxBatchSize = config.BatchPoster.MaxSize
+		}
+
+		log.Info("Creating AnyTrust DA provider", "batchPosterEnabled", config.BatchPoster.Enable)
+
 		// Create AnyTrust factory
-		daFactory, err := factory.NewDAProviderFactory(
-			factory.ModeAnyTrust,
+		daFactory := das.NewFactory(
 			&config.DataAvailability,
-			nil, // referencedaCfg
 			dataSigner,
 			l1client,
 			l1Reader,
 			deployInfo.SequencerInbox,
 			config.BatchPoster.Enable,
 		)
-		if err != nil {
+		log.Info("Created AnyTrust DA factory")
+
+		if err := daFactory.ValidateConfig(); err != nil {
 			return nil, nil, nil, err
 		}
 
-		if err := daFactory.ValidateConfig(); err != nil {
+		var localCleanupFuncs []func()
+		reader, readerCleanup, err := daFactory.CreateReader(ctx)
+		if err != nil {
 			return nil, nil, nil, err
 		}
+		if readerCleanup != nil {
+			localCleanupFuncs = append(localCleanupFuncs, readerCleanup)
+		}
 
-		// Create writer if batch poster is enabled
-		var writerCleanup func()
+		var writer daprovider.Writer
 		if config.BatchPoster.Enable {
-			anytrustWriter, writerCleanup, err = daFactory.CreateWriter(ctx)
+			var writerCleanup func()
+			writer, writerCleanup, err = daFactory.CreateWriter(ctx)
 			if err != nil {
 				return nil, nil, nil, err
 			}
-			withDAWriter = true
+			if writerCleanup != nil {
+				localCleanupFuncs = append(localCleanupFuncs, writerCleanup)
+			}
 		}
 
-		// Create reader
-		var readerCleanup func()
-		anytrustReader, readerCleanup, err = daFactory.CreateReader(ctx)
-		if err != nil {
-			return nil, nil, nil, err
+		headerBytes := daFactory.GetSupportedHeaderBytes()
+
+		// Add AnyTrust writer to writers array if batch poster is enabled
+		if config.BatchPoster.Enable && writer != nil {
```

Comment:
this can go into the above if config.BatchPoster.Enable

Reply:
Fixed in 1e04eb174